app/models/assessment.rb

```diff
@@ -486,16 +485,18 @@ def deserialize(s)
       raise "Name in yaml (#{s['general']['name']}) doesn't match #{name}"
     end
 
-    if s["dates"] && s["dates"]["start_at"] && s["dates"]["due_at"] && s["dates"]["end_at"] &&
-       s["dates"]["visible_at"] && s["dates"]["grading_deadline"]
-      self.due_at = Time.zone.parse(s["dates"]["due_at"])
-      self.start_at = Time.zone.parse(s["dates"]["start_at"])
-      self.end_at = Time.zone.parse(s["dates"]["due_at"])
-      self.visible_at = Time.zone.parse(s["dates"]["visible_at"])
-      self.grading_deadline = Time.zone.parse(s["dates"]["grading_deadline"])
+    if s["dates"] && s["dates"]["start_at"]
+      if s["dates"]["due_at"] && s["dates"]["end_at"] && s["dates"]["grading_deadline"]
+        self.due_at = Time.zone.parse(s["dates"]["due_at"])
+        self.start_at = Time.zone.parse(s["dates"]["start_at"])
+        self.end_at = Time.zone.parse(s["dates"]["due_at"])
```

Comment:
looks like I left a bug here, I saw that due date and end date were the same after importing, this should be `self.end_at = Time.zone.parse(s["dates"]["end_at"])`

Reply:
Fixed